content/docs/learn/typed-errors/from-either-to-raise.md

```diff
@@ -0,0 +1,280 @@
+---
+title: From Either to Raise
+sidebar_position: 5
+description: Learning about Raise from other FP ecosystems.
+---
+
+# From `Either` to `Raise`
+
+Typed errors in other functional ecosystems usually revolve around
+dedicated types like `Either`, which provide the ability to describe
+computations that are successful or end in error. This style is
+fully supported by Arrow, but the DSL based around `Raise` usually
+results in nicer code. This small guide describes common patterns
+in the `Either` style and how they translate into `Raise`.
+
+:::note Working with typed errors
+
+For a more general introduction to typed errors, without assuming
+any prior knowledge on the topic, check [this tutorial](working-with-typed-errors.md).
+
+:::
+
+In the following discussion, we assume the following functions:
+
+```kotlin
+fun f(n: Int): Either<Error, String>
+fun g(s: String): Either<Error, Thing>
+fun h(s: String): Either<Boo, Thing>
+fun Thing.summarize(): String
+```
+
+## Sequential composition
+
+The basic way to compose computations that may error out is by
+sequentially executing each of them, and finishing early if
+an error appears. This is done using `flatMap` — if the next
+computation may also fail — or `map` — if you need to apply
+a pure computation to the result, if available.
+
+Take this possible combination of the aforementioned functions:
+
+```kotlin
+fun foo(n: Int): Either<Error, String> =
+  f(n).flatMap { s ->
+    g(s).map { t ->
+      t.summarize()
+    }
+  }
+```
+
+The translation into the `Raise` DSL describes the same sequence
+of steps, but using a more sequential style:
+
+```kotlin
+fun foo(n: Int): Either<Error, String> = either {
+  val s = f(n).bind()
+  val t = g(s).bind()
+  t.summarize()
+}
+```
+
+There are two important steps in this translation. At the beginning of
+the function, we have the `either` builder, which (1) indicates that
+the result of the function is `Either` of some error type and (2)
+creates a new scope in which you can use the `Raise` DSL.
+
+The second step is that every time that you have an `Either` value
+resulting from another computation (like `f` and `g` above), you need
+to call `bind`. This "injects" any potential error value in the context,
+and continues normal execution otherwise.
```

Comment:
> This "injects" any potential error value in the context

I find that information a bit confusing. What does _context_ mean here? Its the first context is mentioned here and its not clear to me which context you are referring to.

Also I find the word _inject_ misleading as I wouldn't expect an injection to terminate the current calculation.

My suggestion is to rephrase to something like: _This short-circuits the current calculation in case of a potential error or continues with normal execution otherwise_.

Reply:
Thanks for the suggestion, it sounds much more approachable now. I've actually removed any mention of the "inject" word.